.github/workflows/check-winlogbeat.yml

```diff
@@ -0,0 +1,28 @@
+name: check-winlogbeat
+
+on:
+  pull_request:
+    paths:
+      - '.github/workflows/check-winlogbeat.yml'
+      - 'winlogbeat/**'
+      - 'x-pack/winlogbeat/**'
+
+env:
+  BEAT_MODULE: 'winlogbeat'
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Fetch Go version from .go-version
+      run: echo "GO_VERSION=$(cat .go-version)" >> $GITHUB_ENV
+    - uses: actions/setup-go@v3
+      with:
+        go-version: ${{ env.GO_VERSION }}
+    - name: Run check/update
+      run: |
+        export PATH=$PATH:$(go env GOPATH)/bin
```

Comment:
In my experience this hasn't been required. I think `setup-go` is doing this.

Reply:
with go 1.18 I was forced to do it so, I can give a swing to double check if it's still valid

Reply:
https://github.com/elastic/beats/pull/32711/commits/51305cc910e789123bed5567ddb7733a0f3651d2 is the original implementation and https://github.com/elastic/beats/runs/7863034992?check_suite_focus=true is the failed logs

maybe `v3` solves the issue? https://github.com/elastic/beats/pull/32750/commits/33d6734a7e5de36cd3561c8929bbbca44a701b17 is the proposal